backend/dataall/modules/dataset_sharing/services/share_processors/s3_bucket_process_share.py

```diff
@@ -123,6 +126,7 @@ def process_revoked_shares(
         target_environment: Environment,
         source_env_group: EnvironmentGroup,
         env_group: EnvironmentGroup,
+        principal_exist: bool,
```

Comment:
Similar to the comment in the access point shares, I suggest we move the parameter calculation closer to where it is actually used instead of passing it from the top. (check if principal exists directly inside `delete_target_role_bucket_policy`)

Reply:
adressed in [b60f12d](https://github.com/data-dot-all/dataall/pull/1161/commits/b60f12d84c0cef198310dc6b022753d925816b7c)